### PR TITLE
[quant][fx] Add lowering support for functional ops using DefaultNodeQuantizeHandler

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #73436
* #73386
* #73385
* #73233
* __->__ #73120

Summary:
This is to align our implementation with https://github.com/pytorch/rfcs/blob/master/RFC-0019-Extending-PyTorch-Quantization-to-Custom-Backends.md

This changes the ops in DefaultNodeQuantizeHandler to produce reference patterns: dequantize - fp32_op - quantize
and then adds a lowering pass to lower them to quantized ops.

Most of them are straightforward, except for ConvTranspose1d and ConsTranspose2d, which needs reference module support, and we need to add a `from_reference` classmethod in `nnq.ConvTranspose{N}d`

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D34354038](https://our.internmc.facebook.com/intern/diff/D34354038)

### CODE REVIEW

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34354038).

File: torch/ao/quantization/fx/_lower_to_native_backend.py [hunk 1]
```diff
@@ -11,13 +11,27 @@
...
 
+ARGS_TO_SKIP = {
+    torch._ops.ops.quantized.hardswish: ['inplace'],
+    torch._ops.ops.quantized.elu: ['inplace'],
+    torch._ops.ops.quantized.dropout: ['inplace'],
+    torch._ops.ops.quantized.instance_norm:
+    ['running_mean', 'running_var', 'use_input_stats', 'momentum'],
+}
+
+def _is_node(node, modules, func_list, method_list, module_type_list):
```
Comment (vkuzo) [inline torch/ao/quantization/fx/_lower_to_native_backend.py:30]: maybe something like `_is_node_in_list`?

File: torch/ao/quantization/fx/_lower_to_native_backend.py [hunk 2]
```diff
@@ -11,13 +11,27 @@
...
-from ..utils import _parent_name, check_node
+from ..utils import _parent_name
 from ..qconfig import QConfigAny
+from ..quantization_mappings import get_quantized_operator
 from .utils import create_node_from_old_node_preserve_meta
 from typing import Dict, Tuple, Type, List, Callable, Any, Union
 from torch.fx import Node
 import operator
 
+ARGS_TO_SKIP = {
```
Comment (vkuzo) [inline torch/ao/quantization/fx/_lower_to_native_backend.py:22]: maybe `qfun_to_arg_names_to_skip`?

**Review by vkuzo — APPROVED** [review]

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34354038).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34354038).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34354038).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34354038).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34354038).

**Pull request closed without merging**